recording-daemon/mix.c

```diff
@@ -225,6 +247,12 @@ mix_t *mix_new(void) {
 	for (unsigned int i = 0; i < mix_num_inputs; i++)
 		mix->pts_offs[i] = (uint64_t) -1LL;
 
+	for (unsigned int i = 0; i < mix_num_inputs; i++) {
+		// initialise with the first mixer channel to use for each slot. This is set to mix_num_inputs+1
+		// so that we can detect first use and also if the maximum use has been reached.
+		mix->next_idx[i] = mix_num_inputs+1;
```

Comment:
I think this could be made simpler by just initialising each `next_idx[i]` to `i` and then in `mix_get_index()` just get the current value of `next_idx[stream_channel_slot]`, use it as `next`, and then unconditionally increasing it by `mix->channel_slots`. Unless I'm reading the code wrong :grin:

Reply:
Perfect suggestion, this is was I was trying to achieve originally -  but I got lost in over engineering it!